fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -543,8 +543,8 @@ private static void evaluateRequestPreconditions(final Request request,
             etag = new EntityTag(resource.getDescription().getEtagValue());
             date = resource.getDescription().getLastModifiedDate();
         } else {
-            // Use a weak ETag for the LDP-RS
-            etag = new EntityTag(resource.getDescribedResource().getEtagValue(), true);
+            // Use a strong ETag for the LDP-RS when validating If-(None)-Match headers
+            etag = new EntityTag(resource.getDescribedResource().getEtagValue());
```

Comment:
No, it doesn't. And this brings the Fedora API in line with the [HTTP spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html#sec13.3.3). This section of code relates to evaluating (rather than generating) ETags. And for `GET` requests (according to the spec), ETags with the same opaque identifier are considered equal regardless of whether they are strong or weak. However, for requests that initiate changes to a server resource, the spec is clear that the ETags must be strong in order to evaluate to equal. In that case, we were comparing a strong ETag with a weak ETag, and it resulted in a non-match (unlike the _match_ in a read context):

> Strong validators are usable in any context. Weak validators are only usable in contexts that do not depend on exact equality of an entity. For example, either kind is usable for a conditional GET of a full entity. However, only a strong validator is usable for a sub-range retrieval, since otherwise the client might end up with an internally inconsistent entity.